base_layer/core/src/blocks/aggregated_body.rs

```diff
@@ -69,6 +69,27 @@ impl AggregateBody {
         }
     }
 
+    /// Provide read-only access to the input list
+    pub fn inputs(&self) -> &Vec<TransactionInput> {
+        &self.inputs
+    }
+
+    /// Provide read-only access to the output list
+    pub fn outputs(&self) -> &Vec<TransactionOutput> {
+        &self.outputs
+    }
+
+    /// Provide read-only access to the kernel list
+    pub fn kernels(&self) -> &Vec<TransactionKernel> {
+        &self.kernels
+    }
+
+    /// Available for tests only. Get a mutable reference to the inputs
+    #[cfg(test)]
```

Comment:
I'd love to **NOT** have this function (at least it's test-only), so open to suggestions for removing its requirement at (1)

Reply:
Afraid I cannot think of a good way to not have this to facilitate the tests at (1)